### ISSUE TITLE
503 on deploy, scale fixes

### ISSUE THREAD

**Issue by nicholaswyoung** [deis/workflow#222]
When deploying a fresh copy of an app (example: [untilnow/www](https://github.com/untilnow/www/)), the first deploy doesn't properly boot. 

But `scale` the app up and it works. Scale back down to a single instance, and it still works. See the [demo here](https://asciinema.org/a/82bxvn7qyvaylwft8ze72ne25), or read the [Slack conversation](https://deis.slack.com/archives/community/p1462807783002057).

**TLDR:** Deploy = `503`. Scale up/back down = `200`.

**Comment by helgi** [comment]
If you do this again, can you run `deis ps -a <app_name>` and see what it says the status of the pod is

**Comment by nicholaswyoung** [comment]
After monitoring app startup with `deis ps -a ...`, this doesn't look like a bug at all. I made the assumption that once `git push` exits, that the app was ready to go. On the first deploy, where it has to compile the most data, this assumption is invalid.

Thereafter, deploys are (almost) instantly available, though never slower than the initial.

No bug here. Just a poor understanding of what Deis was doing in the background.

**Comment by helgi** [comment]
It should block the operation until the deploy is ready however we can only properly achieve that if health checks are enabled.

By default Kubernetes will assume things are ready as soon as the Pod is up
